docs/promises/create-account.md

```diff
@@ -0,0 +1,39 @@
+---
+sidebar_position: 3
+---
+
+# Creating Accounts
+
+You might want to create an account from a contract for many reasons. One example:
+You want to [progressively onboard](https://www.youtube.com/watch?v=7mO4yN1zjbs&t=2s) users, hiding the whole concept of NEAR from them at the beginning, and automatically create accounts for them (these could be child accounts of your main contract, such as `user123.some-cool-game.near`).
+
+Since an account with no balance is almost unusable, you probably want to combine this with the token transfer from [the last page](./token-tk.md). Here's how to do it:
+
+```rust
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::{json_types::U128, near_bindgen, Balance, AccountId, Promise};
+
+const INITIAL_BALANCE: Balance = 250_000_000_000_000_000_000_000; // 2.5e23yN, 0.25N
+
+#[near_bindgen]
+#[derive(Default, BorshDeserialize, BorshSerialize)]
+pub struct Contract {}
+
+#[near_bindgen]
+impl Contract {
+    #[private]
+    pub fn create_subaccount(&self, prefix: String) -> Promise {
+        let subaccount_id: AccountId = format!("{}.{}", prefix, env::current_account_id());
```

Comment:
Again I think we would really benefit from moving this to the sdk so that users don't have to reinvent the wheel.

Reply:
@austinabell in that case, do you agree with Willem that it's worth adding some SDK method on `AccountId` to construct a subaccount ID?

Reply:
Not sure, hard to say what a clean API would look like for this, but it would only have to validate the length so would be cheaper than validating otherwise.

Reply:
Ideas for API:

* `env::current_account_id().make_subaccount(prefix)`
* `AccountId::join(prefix, env::current_account_id())`